Tasks/UniversalPackagesV0/universalpublish.ts

```diff
@@ -40,18 +41,44 @@ export async function run(artifactToolPath: string): Promise<void> {
 
         let toolRunnerOptions = artifactToolRunner.getOptions();
 
+        let sessionId: string;
+
         if (feedType === "internal")
         {
             // getting inputs
             serviceUri = tl.getEndpointUrl("SYSTEMVSSCONNECTION", false);
 
             packageName = tl.getInput("packageListPublish");
             feedId = tl.getInput("feedListPublish");
+
             // Setting up auth info
             accessToken = pkgLocationUtils.getSystemAccessToken();
             internalAuthInfo = new auth.InternalAuthInfo([], accessToken);
 
             toolRunnerOptions.env.UNIVERSAL_PUBLISH_PAT = internalAuthInfo.accessToken;
+
+            // creating session
+            const useSessionEnabled = tl.getVariable("Packaging.SavePublishMetadata");
+            if (useSessionEnabled) {
+                let packagingLocation: string;
+                try {
+                    // This call is to get the packaging URI(abc.pkgs.vs.com) which is same for all protocols.
+                    packagingLocation = await pkgLocationUtils.getNuGetUriFromBaseServiceUri(
+                        serviceUri,
+                        accessToken);
+                } catch (error) {
+                    tl.debug(JSON.stringify(error));
+                    packagingLocation = serviceUri;
+                }
+
+                const pkgConn = pkgLocationUtils.getWebApiWithProxy(packagingLocation, accessToken);
+                sessionId = await ProvenanceHelper.GetSessionId(
+                    feedId,
+                    "upack",
```

Comment:
We have decided not to use "upack" in any new code. Please use "universal" instead.

Reply:
will do